sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md

```diff
@@ -38,10 +38,13 @@
+| `spark.cosmos.read.customQuery`                                | None    | When provided the custom query will be processed against the Cosmos endpoint instead of dynamically generating the query via predicate push down. Usually it is recommended to rely on Spark's predicate push down because that will allow to generate the most efficient set of filters based on the query plan. But there are a couple of predicates like aggregates (count, group by, avg, sum etc.) that cannot be pushed down yet (at least in Spark 3.1) - so the custom query is a fallback to allow them to be pushed into the query sent to Cosmos. If specified, with schema inference enabled, the custom query will also be used to infer the schema. |
+| `spark.cosmos.read.maxItemCount`                               | `1000`  | Overrides the maximum number of documents that can be returned for a single query- or change feed request. The default value is `1000` - consider increasing this only for average document sizes significantly smaller than 1KB or when projection reduces the number of properties selected in queries significantly (like when only selecting "id" of documents etc.).                                                                                                                                                                                                                                                                                         |
+| `spark.cosmos.read.maxIntegratedCacheStaleness.inMilliseconds` | None    | Sets the max time window in milliseconds for which query results remain cached in the integrated cache in the dedicated gateway connectivity mode. For more information on `MaxIntegratedCacheStaleness`, please visit this [link](https://learn.microsoft.com/azure/cosmos-db/integrated-cache)                                                                                                                                                                                                                                                                                                                                                                  |
+| `spark.cosmos.read.maxIntegratedCacheStaleness.inHours`        | None    | Sets the max time window in hours for which query results remain cached in the integrated cache in the dedicated gateway connectivity mode. For more information on `MaxIntegratedCacheStaleness`, please visit this [link](https://learn.microsoft.com/azure/cosmos-db/integrated-cache)                                                                                                                                                                                                                                                                                                                                                                         |
```

Comment:
Same documentation update for this as well and for other docs. 
Also, what happens if user sets both of them?